coderd/unhanger/detector.go

```diff
@@ -0,0 +1,318 @@
+package unhanger
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/xerrors"
+
+	"github.com/google/uuid"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/database/db2sdk"
+	"github.com/coder/coder/coderd/database/dbauthz"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/provisionersdk"
+)
+
+const (
+	// HungJobDuration is the duration of time since the last update to a job
+	// before it is considered hung.
+	HungJobDuration = 5 * time.Minute
+
+	// HungJobExitTimeout is the duration of time that provisioners should allow
+	// for a graceful exit upon cancellation due to failing to send an update to
+	// a job.
+	HungJobExitTimeout = 3 * time.Minute
```

Comment:
Suggestion: This would (to me) indicate that this value doesn't have greater purpose than to provide a default, reading the code I was found wondering if this variable _also_ is used in this package and what happens when values are mismatched.

```suggestion
	DefaultHungJobExitTimeout = 3 * time.Minute
```

Reply:
I've improved the comment, but this is not a default this is the upper limit a provisioner job should stay running for after failing to send an update to the server.